src/main/scala/mesosphere/marathon/state/Group.scala

```diff
@@ -152,13 +153,58 @@ object Group extends StrictLogging {
   def validGroup(base: PathId, enabledFeatures: Set[String]): Validator[Group] =
     validator[Group] { group =>
       group.id is validPathWithBase(base)
-      group.apps.values as "apps" is every(
-        AppDefinition.validNestedAppDefinition(group.id.canonicalPath(base), enabledFeatures))
+
+      group.transitiveApps as "apps" is everyApp(
+        AppDefinition.validBasicAppDefinition(enabledFeatures) and isChildOfParentId(group)
+      )
+
       group is noAppsAndPodsWithSameId
       group is noAppsAndGroupsWithSameName
       group is noPodsAndGroupsWithSameName
-      group.groupsById.values as "groups" is every(validGroup(group.id.canonicalPath(base), enabledFeatures))
+      
+      group.transitiveGroupValues as "groups" is every(
+        noAppsAndPodsWithSameId and noAppsAndGroupsWithSameName and noPodsAndGroupsWithSameName
+          and isTrue("Group has to be child of groups with parent id") { childGroup =>
+            if (childGroup.id.parent.isRoot) group.groupsById.contains(childGroup.id)
+            else {
+              group.group(childGroup.id.parent) match {
+                case None => false
+                case Some(parentGroup) => parentGroup.groupsById.contains(childGroup.id)
+              }
+            }
+          }
+      )
+    }
+
+  implicit def everyApp(validator: Validator[AppDefinition]): Validator[Iterable[AppDefinition]] = {
```

Comment:
`Validation.every` outputs the wrong paths. Also, I think it is better to show the actual app id.